src/Bicep.Core/Modules/OciModuleArtifactResult.cs

```diff
@@ -28,13 +28,13 @@ public class OciModuleArtifactResult : OciArtifactResult
 
             // Ignore layers we don't recognize for now.
             var expectedLayerMediaType = BicepModuleMediaTypes.BicepModuleLayerV1Json;
-            var mainLayers = this.Layers.Where(l => l.MediaType.Equals(expectedLayerMediaType, MediaTypeComparison));
+            var mainLayers = this.Layers.Where(l => l.MediaType.Equals(expectedLayerMediaType, MediaTypeComparison)).ToArray();
 
-            this.mainLayer = mainLayers.Count() switch
+            this.mainLayer = mainLayers.Length switch
             {
                 0 => throw new InvalidArtifactException($"Expected to find a layer with media type {expectedLayerMediaType}, but found none.", InvalidArtifactExceptionKind.UnknownLayerMediaType),
                 1 => mainLayers.Single(),
-                _ => throw new InvalidArtifactException($"Did not expect to find multiple layer media types of {string.Join(", ", mainLayers.Select(l => l.MediaType).Order().Distinct().ToArray())}", InvalidArtifactExceptionKind.UnknownLayerMediaType)
+                _ => throw new InvalidArtifactException($"Did not expect to find multiple layer media types of {string.Join(", ", mainLayers.Select(l => l.MediaType).Order().Distinct())}", InvalidArtifactExceptionKind.UnknownLayerMediaType)
```

Comment:
Is this correct?  Pretty sure string.Join() requires an array, and array.Distinct() will return an enumerable, no?

Reply:
There is an overload that takes an enumerable: https://learn.microsoft.com/en-us/dotnet/api/system.string.join?view=net-8.0#system-string-join(system-string-system-collections-generic-ienumerable((system-string))).